backend/src/main/scala/cromwell/backend/wdl/OutputEvaluator.scala

```diff
@@ -40,9 +41,17 @@ object OutputEvaluator {
       }
 
       // Attempt to evaluate the expression using all known values
-      def evaluateOutputExpression: OutputResult[WomValue] = {
-        EitherT { Try(output.expression.evaluateValue(allKnownValues, ioFunctions)).map(_.toEither) }
-      }
+      def evaluateOutputExpression: OutputResult[WomValue] = EitherT{ Try{
+        output.expression.evaluateValue(allKnownValues, ioFunctions).toEither.map{
+
+          //If the string contained in a filename is actually a glob,
+          // The output of this expression is expected to be glob-md5(fileName).list
+          case WomSingleFile(fileName) if fileName.contains("*") =>
```

Comment:
I'm curious as why we need this, I would have thought that the fact that the expression contains a glob had already been handled in the `evaluateValue`, so by the time we get here we don't have to do any "is this a glob" checks. Also treating any filename with * as a glob might have unintended consequences as I'm sure someone somewhere named their file `my_awesome_file_with_a_*_in_the_name.boom`

Reply:
I had originally written this within `evaluateValue`, but it is really custom to cromwell impl so it seemed like it belonged here.

Part of the reason for the talk tomorrow is that I am not particularly happy with this solution and I need some fresh eyes on this problem.